README.md

```diff
@@ -0,0 +1,400 @@
+
+``` HTTP
+status: 201
+body:
+
+{
+    "message": "Austin, TX, USA has been added to your favorites."
+}
+```
+
+In the event that the request is unsuccessful, the application will return a 401 status code and a message indicating the request failed (either due to failure to provide an API key or location).
+
+``` HTTP
+status: 401
+body:
+
+{
+    "message": "Unable to add location to favorites."
+}
+```
+
+###### Retrieving favorites
+
+The application supports returning a list of favorite cities registered for a user by submitting a `GET` request to the endpoint `/api/v1/favorites`. This request must include an API key in the body.
+
+``` HTTP
+GET /api/v1/favorites
+Content-Type: application/json
+Accept: application/json
+
+{
+  "api_key": "1a46d0f47e622496255c3ed9b08fef67"
+}
+```
+
+If successful, the application will return a 200 status along with the list of all the users favorited locations, and the current weather at each location.
+
+``` HTTP
+status: 200
+body:
+
+[
+    {
+        "location": "Austin, TX, USA",
+        "current_weather": {
+            "summary": "Clear",
+            "icon": "clear-night",
+            "precipIntensity": 0,
+            "precipProbability": 0,
+            "temperature": 70.31,
+            "humidity": 0.85,
+            "pressure": 1012.38,
+            "windSpeed": 3.38,
+            "windGust": 8.76,
+            "windBearing": 185,
+            "cloudCover": 0.15,
+            "visibility": 9.41
+        }
+    }
+]
+```
+
+In the event an invalid API key is provided or an API key is not provided, the application will return a 401 status and a message indicating the key was invalid.
+
+``` HTTP
+status: 401
+body:
+
+{
+    "error": "Invalid API Key"
+}
+```
+
+###### Deleting favorites
+
+The application also supports removal of favorite locations via a `DELETE` request to the endpoint `/api/v1/favorites`. The request must include the location to be deleted and an API key.
+
+``` HTTP
+DELETE /api/v1/favorites
+Content-Type: application/json
+Accept: application/json
+
+{
+  "api_key": "1a46d0f47e622496255c3ed9b08fef67",
+  "location": "Austin, TX, USA"
+}
+```
+
+The application will respond with a status of 204 when deleting the favorite.
+
+``` HTTP
+status: 204
+```
+
+In the event a location or API key is not provided, the application returns a 401 status.
+
+``` HTTP
+status: 401
+body:
+
+{
+  "error": "Invalid API Key"
+}
+```
+
+### Known Issues
+
+- The application requires testing backfill
+
+- At this time, the error messaging could be more verbose to better indicate the reason for a bad request on favorite endpoints
+
+- The application should make efforts remove the country abbreviation from some requests
+
+- The application will re-geocode a location if the formatting of a forecast does not match the exact formatting of a resolved address
```

Comment:
Does a location sent with a differently formatted query get favorited twice?

Reply:
Results with a differently formatted query will result in the Geocode call being made multiple times, however only one record is created as the resolved address from the service is being used in place of the user's query for the database record.